src/app/shared/components/template/template-container.component.ts

```diff
@@ -93,6 +97,12 @@ export class TemplateContainerComponent implements OnInit, OnDestroy, ITemplateC
     await this.renderTemplate();
   }
 
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.templatename) {
```

Comment:
Good spot. I've implemented a signla based approach in https://github.com/IDEMSInternational/open-app-builder/pull/2531/commits/f31e5d3ff53f6d83680ec874569692392c556679. I would have liked to have defined `template` as a computed signal that evaluates when `templatename` updates, however I was a bit confused by some of the logic in `renderTemplate()` and `forceRerender()`, and decided it would be best not spend time trying to unpick this as part of this PR necessarily, so I've tried to leave that untouched as much as possible.

Reply:
Yeah at this point I think probably best to leave loading the template as-is. I noticed that we were still getting a double render from the effect and ngOnInit, so I removed the onInit code with [e0679c8](https://github.com/IDEMSInternational/open-app-builder/pull/2531/commits/e0679c86842d3368553625329a94f9b5df7d9613) and also did some minor tidying to make more explicit that signal value passed to the render function (could see you had included a manual workaround to call the signal value in the effect anyways).